packages/hash/frontend/src/pages/[account-slug]/entities/[entity-uuid].page/entity-editor/links-section/link-table/cells/linked-with-cell/entity-selector.tsx

```diff
@@ -0,0 +1,135 @@
+import {
+  Entity,
+  EntityId,
+  EntityTypeWithMetadata,
+} from "@hashintel/hash-subgraph";
+import { getRoots } from "@hashintel/hash-subgraph/src/stdlib/roots";
+import { useEffect, useMemo, useRef, useState } from "react";
+import { useBlockProtocolAggregateEntities } from "../../../../../../../../../components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolAggregateEntities";
+import { generateEntityLabel } from "../../../../../../../../../lib/entities";
+import { HashSelectorAutocomplete } from "../../../../../../../types/entity-type/hash-selector-autocomplete";
+import { useEntityEditor } from "../../../../entity-editor-context";
+
+interface EntitySelectorProps {
+  onSelect: (option: Entity) => void;
+  onCancel: () => void;
+  expectedEntityTypes: EntityTypeWithMetadata[];
+  entityIdsToFilterOut?: EntityId[];
+}
+
+export const EntitySelector = ({
+  onSelect,
+  onCancel,
+  expectedEntityTypes,
+  entityIdsToFilterOut = [],
+}: EntitySelectorProps) => {
+  const { entitySubgraph } = useEntityEditor();
+  const { aggregateEntities } = useBlockProtocolAggregateEntities();
+  const [search, setSearch] = useState("");
+
+  const [entities, setEntities] = useState<Entity[]>([]);
+  const [loading, setLoading] = useState(true);
+  const highlightedRef = useRef<null | Entity>(null);
+
+  useEffect(() => {
+    const init = async () => {
+      try {
+        setLoading(true);
+        const { data } = await aggregateEntities({
+          data: {
+            rootEntityTypeIds: expectedEntityTypes.map(
+              ({ schema }) => schema.$id,
+            ),
+          },
+        });
+
+        if (data) {
+          setEntities(getRoots(data));
+        }
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    void init();
+  }, [aggregateEntities, expectedEntityTypes]);
+
+  const sortedAndFilteredEntities = useMemo(() => {
+    return [...entities]
+      .filter(
+        (entity) =>
+          !entityIdsToFilterOut.includes(entity.metadata.editionId.baseId),
+      )
+      .sort((a, b) =>
+        a.metadata.editionId.baseId.localeCompare(b.metadata.editionId.baseId),
+      );
+  }, [entities, entityIdsToFilterOut]);
```

Comment:
On this PR, only new part to this code it this filter & sort part. Rest of this file is copy-paste from it's previous location.